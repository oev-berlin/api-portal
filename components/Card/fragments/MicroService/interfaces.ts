interface MicroServiceProps {
    serviceName: string,
    isMicroservice: boolean,
}

export type { MicroServiceProps };
