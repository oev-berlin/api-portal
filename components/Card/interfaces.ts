import { ProjectData } from '../../utils/interfaces';

interface CardProps{
    projectData: ProjectData
}

export type { CardProps };
