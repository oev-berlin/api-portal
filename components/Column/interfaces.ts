import {projectData} from "../../utils/interfaces";

interface columnData {
    name: string,
    projects: projectData[],
}

export {columnData};