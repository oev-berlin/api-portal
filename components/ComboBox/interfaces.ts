export interface ComboBoxProps{
    searchOptions: string[],
}
