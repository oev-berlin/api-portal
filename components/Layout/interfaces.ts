import React from 'react';

export interface LayoutProps {
    searchOptions: string[],
    children: React.ReactNode,
}
