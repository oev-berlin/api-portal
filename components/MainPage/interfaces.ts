type docType = "backend" | "microservice";

interface projectData {
    id: string,
    name: string,
    description: string,
    swaggerInformation: object,
    docsType: docType,
    microservices: string[],
    externalServices: string[]
}

interface columnData {
    name: string,
    projects: JSON[]
}

export {projectData, columnData};

