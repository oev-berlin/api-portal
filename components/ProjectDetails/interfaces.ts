interface ProjectDetailsProps {
    name: string|undefined,
    description?: string|undefined,
}

export type { ProjectDetailsProps };
