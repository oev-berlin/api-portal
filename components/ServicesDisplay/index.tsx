import { ServicesDisplay } from './ServicesDisplay';

export { ServicesDisplay };
