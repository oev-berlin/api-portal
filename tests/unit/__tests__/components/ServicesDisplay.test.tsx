import { render, cleanup } from '@testing-library/react';
import { toHaveNoViolations } from 'jest-axe';
import React from 'react';
import 'jest-styled-components';
import { ServicesDisplay } from '../../../../components/ServicesDisplay';
import { projectInfo } from '../../../../utils/interfaces';
import exampleProject from '../../../../public/apis_docs/example_1.json';

expect.extend(toHaveNoViolations);

const testProject: projectInfo = exampleProject.info;
const setupComponent = ({ title, services }: { title: string, services: string[] }) => render(
  <ServicesDisplay title={title} services={services} />,
);
describe('ServicesDisplay', () => {
  afterEach(() => {
    cleanup();
  });

  describe('UI Tests', () => {
    it('ServicesDisplay component (type internal, title and services are not empty) should render without crashing', async () => {
      const screen = setupComponent({
        title: 'Internal Services',
        services: testProject.microservices,
      });
      expect(screen.container).toBeDefined();
    });
    it('ServicesDisplay component (type external, title and services are not empty) should render without crashing', async () => {
      const screen = setupComponent({
        title: 'External Services',
        services: testProject.externalservices,
      });
      expect(screen.container).toBeDefined();
    });
    it('ServicesDisplay component (type internal, title is empty) should render without crashing', async () => {
      const screen = setupComponent({
        title: '',
        services: testProject.microservices,
      });
      expect(screen.container).toBeDefined();
    });
    it('ServicesDisplay component (type external, title is empty) should render without crashing', async () => {
      const screen = setupComponent({
        title: '',
        services: testProject.externalservices,
      });
      expect(screen.container).toBeDefined();
    });
    it('ServicesDisplay component (Services is empty) should render without crashing', async () => {
      const screen = setupComponent({
        title: 'Internal Services',
        services: [],
      });
      expect(screen.container).toBeDefined();
    });
    it('ServicesDisplay component (title and services are empty) should render without crashing', async () => {
      const screen = setupComponent({
        title: '',
        services: [],
      });
      expect(screen.container).toBeDefined();
    });
  });

  it('ServicesDisplay component should render correctly', async () => {
    const screen = setupComponent({ title: 'Internal Services', services: testProject.microservices });
    await screen
      .findByText(/internal services/i)
      .then((element) => expect(element).toBeInTheDocument());
    await screen
      .findByText(/mail-service/)
      .then((element) => expect(element).toBeInTheDocument());
    await screen
      .findByText(/pdf-service/i)
      .then((element) => expect(element).toBeInTheDocument());
  });

  describe('Snapshots Tests', () => {
    it('should match a basic snapshot (type internal, title and services are not empty)', () => {
      const { container } = setupComponent({ title: 'Internal Services', services: testProject.microservices });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (type external, title and services are not empty)', () => {
      const { container } = setupComponent({
        title: 'External Services',
        services: testProject.externalservices,
      });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (type internal, title is empty)', () => {
      const { container } = setupComponent({ title: '', services: testProject.microservices });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (type external, title is empty)', () => {
      const { container } = setupComponent({ title: '', services: testProject.externalservices });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (services is empty)', () => {
      const { container } = setupComponent({ title: 'External Services', services: [] });
      expect(container).toMatchSnapshot();
    });
    it('should match a basic snapshot (title and services are empty)', () => {
      const { container } = setupComponent({ title: '', services: [] });
      expect(container).toMatchSnapshot();
    });
  });
});
