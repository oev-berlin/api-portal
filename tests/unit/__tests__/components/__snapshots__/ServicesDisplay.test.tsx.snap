// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesDisplay Snapshots Tests should match a basic snapshot (services is empty) 1`] = `
.c1 {
  margin: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  text-align: left;
  margin-left: 2rem;
}

<div>
  <h2
    class="c0"
  >
    External Services
  </h2>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 c1 css-ee20y-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
    />
  </div>
</div>
`;

exports[`ServicesDisplay Snapshots Tests should match a basic snapshot (title and services are empty) 1`] = `
.c1 {
  margin: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  text-align: left;
  margin-left: 2rem;
}

<div>
  <h2
    class="c0"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 c1 css-ee20y-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
    />
  </div>
</div>
`;

exports[`ServicesDisplay Snapshots Tests should match a basic snapshot (type external, title and services are not empty) 1`] = `
.c1 {
  margin: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  text-align: left;
  margin-left: 2rem;
}

.c3 {
  text-align: center;
  padding: 0.5rem;
}

<div>
  <h2
    class="c0"
  >
    External Services
  </h2>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 c1 css-ee20y-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          WRZRechner
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ServicesDisplay Snapshots Tests should match a basic snapshot (type external, title is empty) 1`] = `
.c1 {
  margin: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  text-align: left;
  margin-left: 2rem;
}

.c3 {
  text-align: center;
  padding: 0.5rem;
}

<div>
  <h2
    class="c0"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 c1 css-ee20y-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          WRZRechner
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ServicesDisplay Snapshots Tests should match a basic snapshot (type internal, title and services are not empty) 1`] = `
.c1 {
  margin: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  text-align: left;
  margin-left: 2rem;
}

.c3 {
  text-align: center;
  padding: 0.5rem;
}

<div>
  <h2
    class="c0"
  >
    Internal Services
  </h2>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 c1 css-ee20y-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          email-service
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          PDF Service
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          Email Service
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ServicesDisplay Snapshots Tests should match a basic snapshot (type internal, title is empty) 1`] = `
.c1 {
  margin: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 {
  text-align: left;
  margin-left: 2rem;
}

.c3 {
  text-align: center;
  padding: 0.5rem;
}

<div>
  <h2
    class="c0"
  />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 c1 css-ee20y-MuiPaper-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          email-service
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          PDF Service
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
        >
          Email Service
        </div>
      </div>
    </div>
  </div>
</div>
`;
